src/libsyntax_ext/deriving/mod.rs

```diff
@@ -104,13 +105,40 @@ pub fn expand_derive(cx: &mut ExtCtxt,
         }
     };
 
-    if mitem.value_str().is_some() {
-        cx.span_err(mitem.span, "unexpected value in `derive`");
+    let mut derive_attrs = Vec::new();
+    item = item.map_attrs(|attrs| attrs.into_iter().filter_map(|attr| {
```

Comment:
Could this use [`Iterator#partition`](https://doc.rust-lang.org/nightly/std/iter/trait.Iterator.html#method.partition) instead?